scroll-search.lic

```diff
@@ -0,0 +1,533 @@
+	  return
+	end
+
+	echo("Scroll type count is: #{scroll_types.length}.") if $debug_mode
+
+	# Clear current scroll listing from owned scrolls if
+	# user is searching an existing container
+	if @owned_scrolls[$container_full]
+	  echo("Clearing scroll array for this container, since it was previously searched --- #{$container_full}!") if $debug_mode
+	  @owned_scrolls[$container_full] = []
+	end
+	
+	# reset scroll count
+	@scroll_count = 0
+	
+	scroll_types.each do |scroll|
+	  echo "Searching for scrolls of type #{scroll}." if $debug_mode
+
+	  scroll_list = get_scrolls(scroll)
+	  next if scroll_list.empty?
+
+	  look_scrolls(scroll_list)
+	end
+
+	# Save scrolls to user variables.
+	if UserVars.owned_scrolls.nil? && @owned_scrolls.nil?
+	  echo("Creating an empty hash for vars!") if $debug_mode
+	  UserVars.owned_scrolls = {}
+	  return
+	end
+	
+	existing_hash = UserVars.owned_scrolls ? UserVars.owned_scrolls.clone : @owned_scrolls.clone
+	new_hash = @owned_scrolls
+	existing_hash.merge!(new_hash) 
+
+	UserVars.owned_scrolls = new_hash
+	UserVars.scroll_counter = @global_scroll_index
+	
+	echo ("Total scrolls added: #{@owned_scrolls.length}!") if $debug_mode
+
+	# Empty your hands
+	empty_hands
+
+	echo_message("SEARCH COMPLETED")
+  end
+
+  def display_scrolls()
+	current_scrolls = UserVars.owned_scrolls
+
+	if current_scrolls.length <= 0 || current_scrolls.nil?
+	  echo_message("NO SCROLLS TO DISPLAY -- SEARCH FIRST!")
+	  return
+	end
+
+	# Formatted Scroll listing
+	echo("|#{' Scroll List '.center(73,'=')}|")
+	echo("|#{'#'.center(5)}|#{'Scroll Name'.center(20)}|#{'Container'.center(30)}|#{'Type'.center(15)}|")
+	echo("|#{''.center(73,'=')}|")
+
+	# Iterate over scrolls to display
+	current_scrolls.each do |container, scroll_array|
+	  scroll_array.each_with_index do |scroll_info, index|
+		echo_scroll_list_line(scroll_info.index, scroll_info.spell, scroll_info.container, scroll_info.get_type)
+	  end
+	end
+
+	echo("|#{''.center(73,' ')}|")
+	echo("|#{'(;send <get #> or ;send <transfer # to <container>>)'.center(73,' ')}|")
+	echo("|#{''.center(73,'=')}|")
+
+	line = get until line.strip =~ /^get\s+(\d+)$|^tran.+\s+(\d+)\s+to\s+(\w+)$/i
+
+	case line
+	when /^get\s+(\d+)$/i
+	  scroll_index = Regexp.last_match(1)
+	  echo("Scroll index is #{scroll_index}.") if $debug_mode
+
+	  grab_scroll(scroll_index)
+	when /^tran.+\s+(\d+)\s+to\s+(\w+)$/i
+	  scroll_index = Regexp.last_match(1)
+	  transfer_container = Regexp.last_match(2)
+
+	  echo("Index is #{scroll_index}. Transfer is: #{transfer_container}") if $debug_mode
+
+	  transfer_scroll(scroll_index, transfer_container)
+	end
+  end
+
+  # Private methods
+  private
+
+  def get_scrolls(scroll)
+	scroll_noun = scroll #get_noun(scroll)
+	parameter = "C #{scroll_noun}"
+	scrolls = rummage(parameter, $container)
+	return scrolls
+  end
+
+  def look_scrolls(all_scrolls)
+
+	all_scrolls.each_with_index do |scroll, index|
```

Comment:
Doesn't DR (the game itself) only support up to "twelve" for an ordinal? Have you tested this with thirteen scrolls in a container?